core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -539,41 +534,60 @@ private ExecutionModel parseCreateTable(GenericLexer lexer, SqlExecutionContext
+        int maxUncommittedRows = configuration.getMaxUncommittedRows();
+        long o3MaxLag = configuration.getO3MaxLag();
+
+        if (tok != null && isWithKeyword(tok)) {
+            ExpressionNode expr;
+            while ((expr = expr(lexer, (QueryModel) null)) != null) {
+                if (Chars.equals(expr.token, '=')) {
+                    if (isMaxUncommittedRowsKeyword(expr.lhs.token)) {
+                        try {
+                            maxUncommittedRows = Numbers.parseInt(expr.rhs.token);
+                        } catch (NumericException e) {
+                            throw SqlException.position(lexer.getPosition()).put(" could not parse maxUncommittedRows value \"").put(expr.rhs.token).put('"');
                         }
-                        tok = optTok(lexer);
-                        if (null != tok && Chars.equals(tok, ',')) {
-                            continue;
+                    } else if (isO3MaxLagKeyword(expr.lhs.token)) {
+                        o3MaxLag = SqlUtil.expectMicros(expr.rhs.token, lexer.getPosition());
+                    } else {
+                        throw SqlException.position(lexer.getPosition()).put(" unrecognized ").put(expr.lhs.token).put(" after WITH");
+                    }
+                    tok = optTok(lexer);
+                    if (null != tok && Chars.equals(tok, ',')) {
+                        CharSequence peek = optTok(lexer);
+                        if (peek != null && isInKeyword(peek)) { // in volume
+                            tok = peek;
+                            break;
                         }
-                        break;
+                        lexer.unparseLast();
+                        continue;
                     }
-                    throw SqlException.position(lexer.getPosition()).put(" expected parameter after WITH");
+                    break;
                 }
+                throw SqlException.position(lexer.getPosition()).put(" expected parameter after WITH");
             }
         }
-
         model.setMaxUncommittedRows(maxUncommittedRows);
         model.setO3MaxLag(o3MaxLag);
-        final boolean isWalEnabled =
-                configuration.isWalSupported() && PartitionBy.isPartitioned(model.getPartitionBy()) && (
-                        (walSetting == WAL_NOT_SET && configuration.getWalEnabledDefault()) || walSetting == WAL_ENABLED
-                );
 
-        model.setWalEnabled(isWalEnabled);
+        if (tok != null && isInKeyword(tok)) {
+            tok = tok(lexer, "volume");
+            if (!isVolumeKeyword(tok)) {
+                throw SqlException.position(lexer.getPosition()).put("expected 'volume'");
+            }
+            tok = tok(lexer, "path for volume");
+            if (Os.isWindows()) {
+                throw CairoException.critical(0).position(lexer.getPosition()).put("'in volume' is not supported in Windows");
```

Comment:
nit: we already use "not supported on Windows" in other error messages:
```suggestion
                throw CairoException.critical(0).position(lexer.getPosition()).put("'in volume' is not supported on Windows");
```